src/object/srv_obj.c

```diff
@@ -1756,14 +1756,17 @@ obj_ioc_init(uuid_t pool_uuid, uuid_t coh_uuid, uuid_t cont_uuid, int opc,
 }
 
 static void
-obj_ioc_fini(struct obj_io_context *ioc)
+obj_ioc_fini(struct obj_io_context *ioc, int err)
 {
 	if (ioc->ioc_coh != NULL) {
 		ds_cont_hdl_put(ioc->ioc_coh);
 		ioc->ioc_coh = NULL;
 	}
 
 	if (ioc->ioc_coc != NULL) {
+		if (obj_is_modification_opc(ioc->ioc_opc) && err == 0 &&
+		    daos_oclass_is_ec(&ioc->ioc_oca))
+			ds_cont_ec_timestamp_update(ioc->ioc_coc);
```

Comment:
why not use the epoch associated with the update?   Do we not have that information?

Reply:
hmm, use the update epoch is a bit racey.  see this case.

1. get update RPC and set update epoch.
2. yield, for example get bulk data.
3. start ec aggregation, finish aggregation, and update ec hae.
4. resume, finish update, then update ec timestamp with the hlc() in 1. which is older than ec hae.

Reply:
Since aggregation uses epoch, I think we should use the epoch from modify RPC instead of current HLC. I don't quite follow the race you described.

Reply:
modify RPC might have earlier epoch than the EC aggregation hae, due to the case I described above.  Then EC aggregation will not be scheduled.

>Since aggregation uses epoch, I think we should use the epoch from modify RPC 

Hmm, why?  I do not think they are directly related.

Reply:
How can modify RPC have earlier epoch than HAE? The incoming epoch must be greater than the aggregation upper bound, so it must be greater than HAE, right?

Reply:
Discussed with Di offline, I'm fine with using this approximate value instead of the accurate epoch from RPC.